Sources/StreamChat/WebSocketClient/EventMiddlewares/ThreadReadUpdaterMiddleware.swift

```diff
@@ -0,0 +1,25 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct ThreadReadUpdaterMiddleware: EventMiddleware {
+    func handle(event: Event, session: DatabaseSession) -> Event? {
+        switch event {
+        case let event as MessageReadEventDTO:
+            if let threadDetails = event.payload.threadDetails {
+                session.markThreadAsRead(parentMessageId: threadDetails.parentMessageId, userId: event.user.id, at: event.createdAt)
+            }
+        case let event as NotificationMarkUnreadEventDTO:
+            // TODO: At the moment, the thread is not returned in the event,
```

Comment:
Is the backend aware of it? Also, good if you avoid using "I" in comments. After checking with the backend, you can rephrase this like "According to the backend, if lastReadMessageId is not present, it means it is a thread".

Reply:
"I" as in the comment "I found that if...". We should avoid using it in comments, since it doesn't help.

Reply:
Ah I see, I'll update it 👍